### PR TITLE
Introducing waf version struct method to check for custom emptiness

### PR DESCRIPTION
**Pull request by oscarDovao**
This PR contains a new waf version struct method that checks for emptiness. Empty, in this case, means that non of the configuration fields are populated. This method can be used to prevent sending a back-end server request which will result on an error

### CODE REVIEW

File: fastly/waf_version.go [hunk 1]
```diff
@@ -259,6 +260,16 @@ type UpdateWAFVersionInput struct {
 	XSSScoreThreshold                int    `jsonapi:"attr,xss_score_threshold,omitempty"`
 }
 
+// IsEmpty checks that UpdateWAFVersionInput is empty, which means that it has no configuration fields populated.
+// if UpdateWAFVersionInput is updated to have a slice this method will not longer work as it is.
```
Comment (pegerto) [inline fastly/waf_version.go]: this comment should explain the  developer decided to use strict comparison with the == operator vs DeepEquals

File: fastly/waf_version.go [hunk 2]
```diff
@@ -259,6 +260,16 @@ type UpdateWAFVersionInput struct {
...
+func (i UpdateWAFVersionInput) IsEmpty() bool {
```
Comment (pegerto) [inline fastly/waf_version.go]: [Minor] I will suggest changing the name of the function form `IsEmpty` that points to an empty structure to `HasChanges` as a suggestion, it changes the meaning of the method as well.?

File: fastly/waf_version.go (hunk 1 above)
Comment (oscarDovao) [reply fastly/waf_version.go]: Included in the docs

File: fastly/waf_version.go (hunk 2 above)
Comment (oscarDovao) [reply fastly/waf_version.go]: Make sense, renamed

**Pull request closed without merging**